core/src/main/java/io/temporal/samples/hello/HelloUpdate.java

```diff
@@ -0,0 +1,281 @@
+     */
+    @WorkflowMethod
+    List<String> getGreetings();
+
+    /*
+     * Define the workflow waitForName update method. This method is executed when the workflow
+     * receives an update request.
+     */
+    @UpdateMethod()
+    Integer waitForName(String name);
+
+    /*
+     * Define an optional workflow update validator. The validator must take the same parameters as the update handle.
+     * The validator is run before the update handle.
+     * If the validator fails by throwing any exception the update request will be rejected and the handle will not run.
+     */
+    @UpdateValidatorMethod(updateName = "waitForName")
+    void waitForNameValidator(String name);
+
+    // Define the workflow exit signal method. This method is executed when the workflow receives a
+    // signal.
+    @SignalMethod
+    void exit();
+  }
+
+  // Define the workflow implementation which implements the getGreetings workflow method.
+  public static class GreetingWorkflowImpl implements GreetingWorkflow {
+
+    // messageQueue holds up to 10 messages (received from updates)
+    List<String> messageQueue = new ArrayList<>(10);
+    List<String> receivedMessages = new ArrayList<>(10);
+    boolean exit = false;
+
+    private final HelloActivity.GreetingActivities activities =
+        Workflow.newActivityStub(
+            HelloActivity.GreetingActivities.class,
+            ActivityOptions.newBuilder().setStartToCloseTimeout(Duration.ofSeconds(2)).build());
+
+    @Override
+    public List<String> getGreetings() {
+
+      while (true) {
+        // Block current thread until the unblocking condition is evaluated to true
+        Workflow.await(() -> !messageQueue.isEmpty() || exit);
+        if (messageQueue.isEmpty() && exit) {
+          // no messages in queue and exit signal was sent, return the received messages
+          return receivedMessages;
+        }
+        String message = messageQueue.remove(0);
+        receivedMessages.add(message);
+      }
+    }
+
+    @Override
+    public Integer waitForName(String name) {
+      if (name.isEmpty()) {
+        // Updates can fail by throwing a TemporalFailure
+        throw ApplicationFailure.newFailure("Cannot greet someone with an empty name", "Failure");
+      }
+      // Updates can mutate workflow state like variables or call activities
+      messageQueue.add(activities.composeGreeting("Hello", name));
+      // Updates can return data back to the client
+      return receivedMessages.size() + messageQueue.size();
+    }
+
+    @Override
+    public void waitForNameValidator(String name) {
+      /*
+       * Update validators have the same restrictions as Queries. So workflow state cannot be
+       * mutated inside a query.
+       */
+      if (receivedMessages.size() >= 10) {
+        /*
+         * Throwing an exception inside an update validator will cause the update to be rejected.
+         * Note: rejected update will not appear in the workflow history
```

Comment:
Maybe a note why `IllegalStateException` can be used here but `IllegalArgumentException` can't be used in the updater